kale/embed/mpca.py

```diff
@@ -93,38 +129,57 @@ def fit(self, X):
         return self
 
     def transform(self, X):
-        """
-        Parameter:
-            X: array-like, shape (I1, I2, ..., n_samples)
-        ----------
-        Return:
-            Transformed data, array-like, shape (i1, i2, ..., n_samples)
+        """Perform dimension reduction on X
+
+        Args:
+            X (ndarray): shape (I1, I2, ..., n_samples)
+
+        Returns:
+            ndarray: Transformed data, shape (i1, i2, ..., n_samples)
         """
         n_spl = X.shape[-1]
         for i in range(n_spl):
             X[..., i] = X[..., i] - self.Xmean
-        return multi_mode_dot(X, self.tPs, modes=[m for m in range(self.n_dim - 1)])
+        return multi_mode_dot(X, self.tPs, modes=[m for m in range(self.ndim - 1)])
 
     def inverse_transform(self, X):
-        """
-        Parameter:
-            X: array-like, shape (i1, i2, ..., n_samples)
-        ----------
-        Return:
-            Data in original shape, array-like, shape (I1, I2, ..., n_samples)
-        """
-        return multi_mode_dot(X, self.tPs, modes=[m for m in range(self.n_dim - 1)], transpose=True)
+        """Transform data in the shape of reduced dimension 
+        back to the original shape
 
+        Args:
+            X (ndarray): shape (i1, i2, ..., n_samples), where i1, i2, ... 
+                are the reduced dimensions of of corresponding mode
+                (1, 2, ...), respectively.
 
-def MPCA_(X, variance_explained=0.97, max_iter=1):
-    """
-    Parameter:
-        X: array-like, ndarray of shape (I1, I2, ..., n_samples)
-        variance_explained: ration of variance to keep (between 0 and 1)
-        max_iter: max number of iteration
-    ----------
-    Return:
-        tPs: list of projection matrices
+        Returns:
+            ndarray: Data in original shape, shape (I1, I2, ..., n_samples)
+        """
+        
+        return multi_mode_dot(X, self.tPs, modes=[m for m in range(self.ndim - 1)], transpose=True)
+
+
+def MPCA_(X, var_explained=0.97, max_iter=1):
+    """MPCA implementation as an independent function
+
+    Args:
+        X (ndarray): training data, shape (I1, I2, ..., n_samples)
```

Comment:
Will *X_train* be better? As in https://scikit-learn.org/stable/auto_examples/applications/plot_face_recognition.html